packages/graphback-core/src/plugin/GraphbackPluginEngine.ts

```diff
@@ -40,12 +40,17 @@ export class GraphbackPluginEngine {
 
     public execute(): GraphbackCoreMetadata {
```

Comment:
This it the core of the plugin. I has some flaws at the moment:

init accepts and caches metadata for create resources, but it might happen that schema changes might affect work we have done in init. 

Generally, I'm thinking that we should get rid of init completely.
Then we can have transformSchema and generateResouces always fetching freshModels (no caching as we do now)


So it should be something like:

```
for (const plugin of this.plugins) {
    this.metadata = plugin.transformSchema(this.metadata);	            
}

for (const plugin of this.plugins) {
   this.metadata = plugin.generateResources(this.metadata);
}
```
WDYT?

Reply:
As for the unit testing issue. Individual plugin can still have plublic methods that we will use for testing (apart from generating resources.